api/awsauth/awsauth/the_registry.py

```diff
@@ -0,0 +1,50 @@
+import inspect
+from awsauth.ses_client import SESClient
+from awsauth.firehose_client import FirehoseClient
+from awsauth.dynamo_client import DynamoDBClient
+from awsauth.ses_client import SESClient
+
+
+class RegistryProvider:
+    @property
+    def ses_client(self) -> SESClient:
+        return SESClient()
+
+    @property
+    def firehose_client(self) -> FirehoseClient:
+        return FirehoseClient()
+
+    @property
+    def dynamodb_client(self) -> DynamoDBClient:
+        return DynamoDBClient()
+
+
+class Registry(object):
+    def __init__(self):
+        self._initialized_objects = {}
+
+    def initialize(self):
+        properties = [
+            prop for prop, obj in inspect.getmembers(self.__class__) if isinstance(obj, property)
```

Comment:
This seems like a lot of logic where a cached_property on a method that creates the desired object would work. It isn't in 3.7 by default but there is https://pypi.org/project/backports.cached-property/

Reply:
Yeah I think that's a good point.  Maybe if we switch to 3.8 or 3.9 we can just use cached property :).  I think I'll keep this way for now.